solvebio/resource/task.py

```diff
@@ -27,3 +27,15 @@ def child_object(self):
     def follow(self):
         """ Follow the child object but do not loop """
         self.child_object.follow(loop=False)
+
+    def cancel(self):
+        """ Cancel a task """
+        if 'cancel' not in self.available_actions:
```

Comment:
The API already validates this and provides a message about the current status as well.